ksqldb-engine/src/main/java/io/confluent/ksql/execution/json/PlanJsonMapper.java

```diff
@@ -45,6 +46,7 @@ public static ObjectMapper create() {
     mapper.enable(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES);
     mapper.enable(DeserializationFeature.FAIL_ON_NULL_CREATOR_PROPERTIES);
     mapper.enable(DeserializationFeature.FAIL_ON_INVALID_SUBTYPE);
+    mapper.setSerializationInclusion(Include.NON_EMPTY);
```

Comment:
This configures Jackson to exclude `null`, `Optional.empty()` and empty collections when serializing.